src/main/java/org/opensearch/neuralsearch/processor/combination/ScoreCombiner.java

```diff
@@ -239,19 +244,26 @@ private List<ScoreDoc> getCombinedScoreDocs(
             if (hitCount == maxHits) {
                 break;
             }
-            scoreDocs.add(getScoreDoc(isSortingEnabled, docId, shardId, combinedNormalizedScoresByDocId, docIdSortFieldMap));
+            scoreDocs.add(getScoreDoc(sort, docId, shardId, combinedNormalizedScoresByDocId, docIdSortFieldMap, isSingleShard));
             hitCount++;
         }
         return scoreDocs;
     }
 
     private ScoreDoc getScoreDoc(
-        final boolean isSortEnabled,
+        final Sort sort,
         final int docId,
         final int shardId,
         final Map<Integer, Float> combinedNormalizedScoresByDocId,
-        final Map<Integer, Object[]> docIdSortFieldMap
+        final Map<Integer, Object[]> docIdSortFieldMap,
+        final boolean isSingleShard
     ) {
+        final boolean isSortEnabled = sort != null;
+        final boolean isSortByScore = isSortOrderByScore(sort);
```

Comment:
isSortByScore method already has a check of sort!=null

Reply:
Therefore, sort!=null will be check twice

Reply:
Yes but we are checking `isSortByScore == false` in the if condition and also we would need `isSortEnabled` in the next if condition